src/devices/Ad7193/Ad7193.cs

```diff
@@ -0,0 +1,838 @@
+    /// </summary>
+    public class Ad7193 : IDisposable
+    {
+        #region Metadata for IDevice
+
+        /// <summary>
+        /// Name of the manufacturer of the device
+        /// </summary>
+        public const string Manufacturer = "Analog Devices";
+
+        /// <summary>
+        /// Name of the product
+        /// </summary>
+        public const string Product = "AD7193";
+
+        /// <summary>
+        /// Category of the device
+        /// </summary>
+        public const string ProductCategory = "ADC";
+
+        /// <summary>
+        /// Description of the device
+        /// </summary>
+        public const string ProductDescription = "4-Channel, 4.8 kHz, Ultralow Noise, 24-Bit Sigma-Delta ADC with PGA";
+
+        /// <summary>
+        /// The URI of the datasheet of the device
+        /// </summary>
+        public const string DataSheetURI = "https://www.analog.com/media/en/technical-documentation/data-sheets/AD7193.pdf";
+        #endregion
+
+        #region Metadata for ISpiDevice
+
+        /// <summary>
+        /// The list of SPI modes that are valid for this device
+        /// </summary>
+        public const SpiMode ValidSpiModes = SpiMode.Mode3;
+
+        /// <summary>
+        /// The maximum frequency that can be used on the SPI bus
+        /// </summary>
+        public const int MaximumSpiFrequency = 10000000;
+        #endregion
+
+        #region metadata for IAdcDevice
+
+        /// <summary>
+        /// Number of ADCs on the device
+        /// </summary>
+        public const int ADCCount = 1;
+
+        /// <summary>
+        /// The maximum bitrate of the ADC
+        /// </summary>
+        public const int ADCBitrate = 24;
+
+        /// <summary>
+        /// The maximum sampling rate of the ADC
+        /// </summary>
+        public const int ADCSamplerate = 4800;
+
+        /// <summary>
+        /// The number of channels the ADC has
+        /// </summary>
+        public const int ADCInputChannelCount = 8;
+        #endregion
+
+        private readonly object _spiTransferLock = new object();
+        private readonly Stopwatch _stopWatch = new Stopwatch();
+
+        private SpiDevice _spiDevice = null;
+
+        /// <summary>
+        /// The list of received ADC values
+        /// </summary>
+        // protected ConcurrentQueue<AdcValue> adcValues = new ConcurrentQueue<AdcValue>();
+        protected BlockingCollection<AdcValue> adcValues = new BlockingCollection<AdcValue>();
+
+        /// <summary>
+        /// The event that is fired every time a new value is received from the ADC
+        /// </summary>
+        public event EventHandler<AdcValueReceivedEventArgs> AdcValueReceived;
+
+        private readonly StringBuilder _sb = new StringBuilder();
+
+        private readonly uint[] _registerCache = { 0x00, 0x080060, 0x000117, 0x000000, 0xa2, 0x00, 0x000000, 0x000000 };
```

Comment:
What is the point of keeping a register Cache? I'm not saying that is wrong, but usually you really want to go and check it everytime from the device itself instead of using something from the cache.

Reply:
There are some rare cases when the cache is very useful: when we are in continuous mode, the device send the ADC values continuously on the bus. If we don't ask the ADC to include the current status register in the results, we can win a little bandwidth, but we don't know on which channel that particular ADC value was converted on.
If we previously selected only one channel, that's now a problem, we know that all values are from channel x. 
In this case we should not break the flow of the data, just because we want to know the channel number. It's alright, if we get it from cache.
The situation is similar to values like the gain and filter, these values are typically set before the start of the continuous conversion and they remain the same for sure until we stop that conversion.